telegram/request/_requestparameter.py

```diff
@@ -127,14 +151,15 @@ def from_input(cls, key: str, value: object) -> 'RequestParameter':
             param_values = []
             input_files = []
             for obj in value:
-                param_value, input_file = cls._value_and_input_file_from_input(obj)
-                param_values.append(param_value)
+                param_value, input_file = cls._value_and_input_files_from_input(obj)
+                if param_value is not None:
+                    param_values.append(param_value)
                 input_files.extend(input_file)
             return RequestParameter(
                 name=key, value=param_values, input_files=input_files if input_files else None
             )
 
-        param_value, input_files = cls._value_and_input_file_from_input(value)
+        param_value, input_files = cls._value_and_input_files_from_input(value)
         return RequestParameter(
             name=key, value=param_value, input_files=input_files if input_files else None
```

Comment:
why not check for `param_value is not None` here?

Reply:
checking it wouldn't change anything. we still need to provide a a `value` parameter - or change the definition of `RequestParameter`.
But I guess the confusion is rather due to the `if param_value is not None` in the `list` case above: That's a bit similar to https://github.com/python-telegram-bot/python-telegram-bot/pull/2731#discussion_r856524189. I.e. the only case where `param_value` could be `None` is if the parameter is an `InputFile`. Currently there doesn't exist a parameter that accepts a list of `InputFiles` and even if there were, those would have to be `attach://` ed (due to the nesting logic)

For both this and your comment above we could maybe

1. Add comments
2. Remove logic for cases that currently can't happen (may make mypy unhappy)
3. raise exceptions for cases that (at least currently) musn't happen
4. a mix of the above
5. leave as is xD

Reply:
I'm fine with leaving as is then, I think understand the file handling part better now